### ISSUE TITLE
Serialized Floats should be Decimals with Scale 4

### ISSUE THREAD

**Issue by lthiery** [helium/packet-purchaser#17]
Semtech's GWMP protocol tends to represent `lsnr` and `freq` as decimals. 

So you might get the following JSON serlialization:

```
{"rxpk":[{"chan":0,"codr":"4/5","data":"QD51kgWAGZoFMahcof/U2VsXpxPsd0Pb4O5M0g==","datr":"SF12BW125","freq":868.3,"lsnr":-16.80,"modu":"LORA","rfch":0,"rssi":-110,"size":28,"stat":1,"time":"1970-02-16T01:13:07Z","tmst":4241221449}]}
```

At some point through the Helium pipeline ( GWMP -> Erlang -> proto -> Erlang -> GWMP), that JSON object could come out like this:

```
{"rxpk":[{"chan":0,"codr":"4/5","data":"QD51kgWAGZoFMahcof/U2VsXpxPsd0Pb4O5M0g==","datr":"SF12BW125","freq":868.2999877929688,,"lsnr":-16.799999237060547,"modu":"LORA","rfch":0,"rssi":-110,"size":28,"stat":1,"time":"1970-02-16T01:13:07Z","tmst":4241221449}]}
```

To keep things simple, I would suggest rounding the float to the closest ~~second~~ fourth decimal when serializing it to JSON again (IN865 has frequencies such as 865.4025 MHz)

**Comment by lthiery** [comment]
This has become a problem when trying to test with Chirpstack. I had a virtual device directly connected to CS with no problem, but when I route via Packet Purchaser, the frequency mangling upsets Chirpstack:

```
time="2021-09-16T15:29:27.61188299Z" level=error msg="uplink: processing uplink frame error" ctx_id=e4c86eff-02fa-4422-8be5-c7b254a5bedf error="run join-response flow error: get rx1 frequency error: lorawan/band: unknown channel for frequency: 905299987"
```

**Comment by lthiery** [comment]
Resolved by #44
